process/receiptslog/receiptsManager.go

```diff
@@ -0,0 +1,65 @@
+package receiptslog
+
+import (
+	"github.com/multiversx/mx-chain-core-go/data/state"
+)
+
+// ArgsReceiptsManager is the structure that holds the components needed to a new receipts manager
+type ArgsReceiptsManager struct {
+	TrieHandler Interactor
+}
+
+// ArgsGenerateReceiptsAndSave is the DTO needed to provided input data to generate receipts
+type ArgsGenerateReceiptsAndSave struct {
+	// add data needed to create receipt structure
+	// logs
+}
+
+type receiptsManager struct {
+	trieInteractor Interactor
+}
+
+// NewReceiptsManager will create a new instance of receipts manager
+func NewReceiptsManager(args ArgsReceiptsManager) (*receiptsManager, error) {
+	if args.TrieHandler == nil {
+		return nil, ErrNilTrieInteractor
+	}
+
+	return &receiptsManager{
+		trieInteractor: args.TrieHandler,
+	}, nil
+}
+
+// GenerateReceiptsTrieAndSaveDataInStorage will generate the receipts trie based on the input data and return the receipt trie root hash
+func (rm *receiptsManager) GenerateReceiptsTrieAndSaveDataInStorage(args ArgsGenerateReceiptsAndSave) ([]byte, error) {
+	// TODO  generate the list of receipts
+
+	err := rm.trieInteractor.CreateNewTrie()
+	if err != nil {
+		return nil, err
+	}
+
+	receipts := make([]state.Receipt, 0)
```

Comment:
so the TODO in this file is related to this this receipts list?

Reply:
yes, this part will be implemented in the next PRs when these components are integrated.